skglm/solvers/cd_solver.py

```diff
@@ -268,19 +269,27 @@ def cd_solver(
         for epoch in range(max_epochs):
             if is_sparse:
                 _cd_epoch_sparse(
-                    X.data, X.indptr, X.indices, y, w, Xw, datafit, penalty,
-                    ws)
+                    X.data, X.indptr, X.indices, y, w[:n_features], Xw,
+                    datafit, penalty, ws)
             else:
-                _cd_epoch(X, y, w, Xw, datafit, penalty, ws)
+                _cd_epoch(X, y, w[:n_features], Xw, datafit, penalty, ws)
+
+            # update intercept
+            if fit_intercept:
+                intercept_old = w[-1]
+                w[-1] -= datafit.intercept_update_step(y, Xw)
+                Xw += (w[-1] - intercept_old)
 
             # 3) do Anderson acceleration on smaller problem
             # TODO optimize computation using ws
```

Comment:
performing acceleration while considering the intercept is not straightforward using the ``AndersonAcceleration`` class.
I disabled it here but marked it as a potential improvement

Reply:
Are you sure it's not straightforward ? The intercept is the last element of `w` so it should be taken into account easily by the accelerator

Reply:
@mathurinm, I mean taking advantage of working sets (accelerating ``w_ws``) while considering intercept in the AndersonAcceleration.

In this setup, we extrapolate w and intercept.

Reply:
you could do something like `ws_ = np.hstack([ws, -1]) if fit_intercept else ws` and pass `w[ws_]`


Changes such as the current one potentially introduce a performance regression, that would need to be tested and this takes time. So I'd try and make the above fix work, to not change existing behavior

Reply:
@Badr-MOUFAD it's one thing that remains to be done IMO. It's already done in multitask_bcd_solver (`ws_ = np.append(ws, -1)`